File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minValue = INT_MAX;
        int maxProfit = 0;
        for(int &price : prices)
        {
            if(price < minValue)
            {
                minValue = price;
            }
            else
            {
                maxProfit = max(maxProfit, price-minValue);
            }
        }
        return maxProfit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0, min = prices[0];
    for(let i = 1; i < prices.length; i++) {
        maxProfit = Math.max(maxProfit, prices[i]-min);     
        min = Math.min(min, prices[i]);  
    }
    return maxProfit;
};